packages/components/src/components/calendar/calendar.tsx

```diff
@@ -101,27 +141,29 @@ const Calendar = React.memo(
                 }
             };
 
-            const updateSelected = (e, type) => {
+            const updateSelected = (e: React.MouseEvent<HTMLSpanElement>, type: moment.unitOfTime.StartOf) => {
                 if (e) e.stopPropagation();
 
                 if (type === 'day') {
                     updateSelectedDate(e);
                     return;
                 }
 
-                const view_map = {
+                const view_map: Record<string, string> = {
                     month: 'date',
                     year: 'month',
                     decade: 'year',
                 };
-                const date = toMoment(calendar_date)
-                    [type === 'decade' ? 'year' : type](e.target.dataset[type].split('-')[0])
-                    .format(date_format);
 
+                let date = '';
+                if (type) {
+                    const selected_date_part = e?.currentTarget?.dataset?.[type]?.split?.('-')?.[0] || 0;
+                    date = getDate(toMoment(calendar_date), type, date_format, +selected_date_part);
+                }
```

Comment:
is it an equivalent replacement of what was here before?
_currently, if the type is  'decade', we do this:_
const date = toMoment(calendar_date)[**'year'**](e.target.dataset[**'decade'**].split('-')[0]).format(date_format);
_and decade is not used in your new code:_
const date = getDate(toMoment(calendar_date), **'year'**, date_format, e?.currentTarget?.dataset?.[**'year'**]?.split?.('-')?.[0] || 0);

Reply:
@maryia-deriv, as seen in the screenshot, there was  a conditional to check whether `type === decade` and then they  replace it with year. So I just went on to replace the usage of `decade` with `year` elsewhere in the code 

<img width="1018" alt="decade" src="https://user-images.githubusercontent.com/104334373/214575315-8e97055f-8ce1-4b38-a5b1-6d61f020da31.png">

Reply:
@maryia-deriv, you're right. However, I'll change it from `decade` to `years`, since `years` is supported  as part of `moment.unitOfTime.StartOf`. Thank you. I had missed this

Reply:
@maryia-deriv, you are right about the `data-decade` I had missed it, I've decided to rename it to `years` however, because `years` is supported as part of `moment.unitOfTime.StartOf` which is used as the `type` for the `parameter` of `toMoment().clone().startOf()`, yet `decade` is not

Reply:
@jim-deriv just to make sure🙂 it's supposed to be `years` exactly, not `year`, as you mentioned, right?

Reply:
Year, because `year` is already used to represent a single year. The `years` here means a range e.g. `2011-2020`